src/utils.ts

```diff
@@ -40,13 +40,21 @@ export function prepareStoreWithDatabase(
         [flagIndex, ...indices].forEach(({ name, keyPath = name, unique }) =>
           idbObjectStore.createIndex(name, keyPath, { unique })
         );
-      } else if (
-        schema.onUpgradeNeeded &&
-        (await schema.onUpgradeNeeded(db, evt))
-      ) {
-        resolve(db);
+      } else if (schema.onUpgradeNeeded) {
+        const migrated = await schema.onUpgradeNeeded(db, evt, req.transaction?.objectStore(store));
+        if (migrated) {
+          console.info(`${dbName}:${store} migrated to ${dbVersion}`);
+        } else {
+          console.info(`${dbName}:${store} migration failed`);
+        }
+      } else if (req.transaction) {
```

Comment:
Have you guys ever managed to get an active "versionchange" transaction at this line? Have you tested it? From my experience, transaction is closed without awaiting for the Promise to be resolved (no async method support)